Foreman 3.16.0 release notes (#2241)
theforeman/theforeman.org: _includes/manuals/3.16/1.2_release_notes.md

* Smart Proxy Userdata API does not implement MAC-based endpoint ([#38241](https://projects.theforeman.org/issues/38241))
* Send HSTS header to satisfy braindead "security" scanners ([#38432](https://projects.theforeman.org/issues/38432))

#### Smart Proxy - DHCP
* Support setting key-algorithm for ISC DHCP ([#36325](https://projects.theforeman.org/issues/36325))

#### Hammer CLI
* Update field documentation to reflect replaced_by translation ([PR #398](https://github.com/theforeman/hammer-cli/pull/398)), [#38427](http://projects.theforeman.org/issues/38427)

#### Hammer CLI - Foreman
* Ldap group membership ([PR #641](https://github.com/theforeman/hammer-cli-foreman/pull/641)), [#38611](http://projects.theforeman.org/issues/38611)
* Remove ovirt ([PR #639](https://github.com/theforeman/hammer-cli-foreman/pull/639)), [#38294](http://projects.theforeman.org/issues/38294)

*A full list of changes in 3.16.0 is available via [Redmine](https://projects.theforeman.org/issues?set_filter=1&sort=id%3Adesc&status_id=closed&f[]=cf_12&op[cf_12]=%3D&v[cf_12]=1964)*

### Contributors

We'd like to thank the following people who contributed to the Foreman {{page.version}} release:

> <!-- update scripts/committers.rb with the correct versions and dates and fill this in -->

Author: The way I ran the script is:
```
bundle exec ruby ./scripts/committers.rb --multi-line 3.15.0 3.16.0-rc1
```

Could anyone please confirm for me that I've used the correct arguments?

Reviewer: Looks reasonable.

* Remove ovirt ([PR #639](https://github.com/theforeman/hammer-cli-foreman/pull/639)), [#38294](http://projects.theforeman.org/issues/38294)

*A full list of changes in 3.16.0 is available via [Redmine](https://projects.theforeman.org/issues?set_filter=1&sort=id%3Adesc&status_id=closed&f[]=cf_12&op[cf_12]=%3D&v[cf_12]=1964)*

### Contributors

We'd like to thank the following people who contributed to the Foreman {{page.version}} release:

<!-- update scripts/committers.rb with the correct versions and dates and fill this in -->
Adam Lazik,
Adam Růžička,
Alin Cociorvan,
Arvind Jangir,
Bastian Blank,
Bastian Schmidt,
Bernhard Suttner,
Chris Roberts,
Cole Higgins,
Corporte Gadfly,
David Ochner,
Eric Helms,
Evgeni Golov,
Ewoud Kohl van Wijngaarden,
Francesco Di Nucci,
Hao Yu,
Ian Ballou,
Jonathon Turel,
Leos Stejskal,
Lucy Fu,
Lukas Jezek,
Lukáš Hellebrandt,
Macley,
Maria Agaphontzev,
Markus Bucher,
Matt Darcy,
Maximilian Kolb,
Nofar Alfassi,
Nomen Nescio,
Odilon Sousa,
Oleh Fedorenko,
Ondřej Gajdušek,
Pablo Fernández Rodríguez,
Partha Aji,
Pat Riehecky,
Ruediger Pluem,
Samir Jha,
Shimon Shtein,
Usman Sunyaev,

> Viliam Krizan

Reviewer 2: The names here render together without whitespace, where in the last release we added bullet points: https://github.com/theforeman/theforeman.org/blob/gh-pages/_includes/manuals/3.15/1.2_release_notes.md?plain=1#L136-L167
Should we keep the bullet points as before, or is it better to keep the names compact?

Author: I've dug out [release notes compiled by you](https://github.com/theforeman/theforeman.org/blob/gh-pages/_includes/manuals/3.5/1.2_release_notes.md?plain=1#L136-L167) :laughing:  and the names are all on one line, same as most of the previous releases. 3.15 was an exception in this.

Therefore, I'd keep it as it is if no one objects.

### Headline features

#### User Interface & Experience
- **All Hosts Page Redesign** - Continued modernization of the All Hosts page with enhanced functionality, including changing owner, location/organization, disassociating VMs, and other improvements.

> - **Job Invocations Detail Page** - Complete overhaul of job execution details with improved performance, better loading, and enhanced user interface

Reviewer 3: Looking at release notes of previous releases, we seem to have omitted things coming from plugins in the past.

Reviewer 4: Yep, that's also why I recommended the All Hosts page bullet only have Foreman things.

Author: That's a good point, I'll remove it.

#### Infrastructure & Platform Updates
- **NodeJS 22 Upgrade** - Updated UI build stack to NodeJS 22, extending lifecycle support to April 2027
- **PatternFly 5 Migration** - Modernization of UI components by upgrading from PatternFly 3 to PatternFly 5 in Foreman core

#### Networking & Connectivity
- **Dual-Stack IPv4/IPv6 Registration Support** - Full support for client registration in dual-stack environments, enabling seamless registration over both IPv4 and IPv6 networks

#### Data & Analytics
- **Insights Vulnerability Service Integration** - Integration of Insights Vulnerability service within Foreman interface

> - **Advisor Recommendations Display** - Integration of Insights Advisor recommendations and details within Foreman interface

Reviewer 3: Afaik this is foreman_rh_cloud plugin, not sure how much this is applicable to community.

Reviewer 2: @jeremylenz mentioned that you can use foreman_rh_cloud with non-Satellite deployments.

Reviewer 2: But, it is a plugin, so perhaps it still makes sense to leave out ?

Reviewer 3: If we're leaving out rex, we should probably leave this one out as well

Reviewer 2: Conversation continued and now it's unclear, let's leave the changelog as it is now without these entries.

Reviewer 4: Regardless of whether there's an upstream use case for Insights, agree that Insights-related features should go in rh_cloud release notes and not Foreman release notes.

Reviewer 2: More conversation happened internally - now we're at "it should work upstream but you do need a Satellite subscription, so you're probably using Satellite anyway".
Just making ends meet, I don't think this fact changes the PR :)

### Headline features

#### User Interface & Experience

> - **All Hosts Page Redesign** - Continued modernization of the All Hosts page with enhanced functionality, eliminating AngularJS pages and improving user experience for managing thousands of hosts

Reviewer 5: @jeremylenz @parthaa

Reviewer 6: I dont think we need to mention about eliminating Angular JS part. Rest seems ok to me.

> - **All Hosts Page Redesign** - Continued modernization of the All Hosts page with enhanced functionality, eliminating AngularJS pages and improving user experience for managing thousands of hosts

Reviewer 4: ```suggestion
- **All Hosts Page Redesign** - Continued modernization of the All Hosts page with enhanced functionality, including changing owner, location/organization, disassociating VMs, and other improvements.
```

> - **Job Invocations Detail Page** - Complete overhaul of job execution details with improved performance, better loading, and enhanced user interface

Reviewer 5: @kmalyjur

#### Infrastructure & Platform Updates
- **NodeJS 22 Upgrade** - Updated UI build stack to NodeJS 22, extending lifecycle support to April 2027

> - **PatternFly 5 Migration** - Modernization of UI components by upgrading from PatternFly 3 to PatternFly 5 in Foreman core

Reviewer 5: We upgraded the Patternfly 4 components in core and plugins to use 5
we still use pf3

Author: @MariaAga How does this sound to you?
```suggestion
- **PatternFly 5 Migration** - Modernization of UI components by upgrading from PatternFly 4 to PatternFly 5 in Foreman core
```

Reviewer 5: ```suggestion
- **PatternFly 5 Migration** - Modernization of UI components by upgrading our PatternFly 4 UI components to PatternFly 5
```
Can be `in Foreman` or `in Foreman core and plugins`

> - **New Installer 1.0 Release** - Major release of updated installation tooling

Author: @MariaAga, can you please help me put together these two sections?

> - **New Installer 1.0 Release** - Major release of updated installation tooling

Reviewer: I think we drop this. We have not released a 1.0 yet.

#### Networking & Connectivity
- **Dual-Stack IPv4/IPv6 Registration Support** - Full support for client registration in dual-stack environments, enabling seamless registration over both IPv4 and IPv6 networks

#### Data & Analytics

> - **Insights Vulnerability Service Integration** - Support for Technology Preview version of Red Hat Insights Vulnerability service

Reviewer: ```suggestion
- **Insights Vulnerability Service Integration** - Integration of Insights Vulnerability service within Foreman interface
```

> - **Advisor Recommendations Display** - Integration of Red Hat Insights Advisor recommendations and details within Foreman interface

Reviewer: ```suggestion
- **Advisor Recommendations Display** - Integration of Insights Advisor recommendations and details within Foreman interface
```

#### Documentation & Tooling
- **Provisioning Documentation Improvements** - Enhanced documentation for provisioning workflows and best practices

### Upgrade warnings

#### Recommended Pre-Upgrade Actions
- **Audit RHV/oVirt Usage** - If using RHV compute resources, plan migration to alternative platforms before upgrading

### Deprecations

- **🚨 RHV/oVirt Support Completely Removed** - Red Hat Virtualization (RHV) and oVirt compute resource support has been fully removed

>   - **Previous Deprecation**: Feature was deprecated in Foreman 3.3

Reviewer 7: 3.15

  - **Impact**: Users with RHV/oVirt compute resources will lose this functionality

>   - **Action Required**: Must migrate to alternative virtualization platforms before upgrading

Reviewer 7: In 3.15 THEY SHOULD run `foreman-rake ovirt:drop`

Now in 3.16, it's automatic and part of the installation. Rake description:

```
    Remove all oVirt / RHV compute resources & profiles from Foreman.
    Hosts associated with these compute resources will be disassociated and kept.
    Host groups with oVirt compute resources & profiles will have their compute resource and compute profile removed.
    Compute profiles, images, and other resources will be removed.
```
